frontend/src/components/shared/StreamlitMarkdown/StreamlitMarkdown.tsx

```diff
@@ -271,4 +304,44 @@ export function LinkWithTargetBlank(props: LinkProps): ReactElement {
   )
 }
 
+export function Heading(props: TestHeaderProps): ReactElement {
+  const { width } = props
+  const { tag, anchor, body } = props.element
+  const isSidebar = React.useContext(IsSidebarContext)
+  // st.header can contain new lines which are just interpreted as new
+  // markdown to be rendered as such.
+  const [heading, ...rest] = body.split("\n")
+
+  return (
+    <div className="stMarkdown" style={{ width }}>
+      <HeadingWithAnchor tag={tag} anchor={anchor}>
+        <RenderedMarkdown
+          source={heading}
+          allowHTML={false}
+          // this is purely an inline string
+          overrideComponents={{
+            p: Fragment,
```

Comment:
@kmcgrady , I think this is what we want?

Reply:
I think so. I wonder whether the `pre` should be fixed to `Fragment` or `CodeTag`.

I _believe_, `pre`s are block elements and `code` is inline, so there are two options:
1. make `pre`s not render as such and use `Fragment`
2. make `pre` render as if it is inline (which I think `CodeTag` is the replacement).

I'm not 100%, and there will need to be some testing.

Reply:
Looks like both render the same thing so I'm just not going to include for now.